tsdb/errors/errors_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2025 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package errors
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMultiError_Is(t *testing.T) {
+	customErr1 := errors.New("test error 1")
+	customErr2 := errors.New("test error 2")
+
+	testCases := map[string]struct {
+		sourceErrors []error
+		target       error
+		is           bool
+	}{
+		"adding a context cancellation doesn't lose the information": {
+			sourceErrors: []error{context.Canceled},
+			target:       context.Canceled,
+			is:           true,
+		},
+		"adding multiple context cancellations doesn't lose the information": {
+			sourceErrors: []error{context.Canceled, context.Canceled},
+			target:       context.Canceled,
+			is:           true,
+		},
+		"adding wrapped context cancellations doesn't lose the information": {
+			sourceErrors: []error{errors.New("some error"), fmt.Errorf("some message: %w", context.Canceled)},
+			target:       context.Canceled,
+			is:           true,
+		},
+		"adding a nil error doesn't lose the information": {
+			sourceErrors: []error{errors.New("some error"), fmt.Errorf("some message: %w", context.Canceled), nil},
+			target:       context.Canceled,
+			is:           true,
+		},
+		"errors with no context cancellation error are not a context canceled error": {
+			sourceErrors: []error{errors.New("first error"), errors.New("second error")},
+			target:       context.Canceled,
+			is:           false,
+		},
+		"no errors are not a context canceled error": {
+			sourceErrors: nil,
+			target:       context.Canceled,
+			is:           false,
+		},
+		"no errors are a nil error": {
+			sourceErrors: nil,
+			target:       nil,
+			is:           true,
+		},
+		"nested multi-error contains customErr1": {
+			sourceErrors: []error{
+				customErr1,
+				NewMulti(
+					customErr2,
+					fmt.Errorf("wrapped %w", context.Canceled),
+				).Err(),
+			},
+			target: customErr1,
+			is:     true,
+		},
+		"nested multi-error contains custom error": {
```

Comment:
[Nit] Judging by the name of the preceding test case, shouldn't it be referred to as `customErr2`?
```suggestion
		"nested multi-error contains customErr2": {
```